ipaserver/install/ipa_restore.py

```diff
@@ -483,10 +481,11 @@ def run(self):
                 if result.returncode != 0:
                     logger.error('Restarting IPA failed: %s', result.error_log)
         finally:
+            tmp_dir = tempfile.gettempdir()
             try:
-                os.chdir(cwd)
+                os.chdir(tmp_dir)
             except Exception as e:
-                logger.error('Cannot change directory to %s: %s', cwd, e)
+                logger.error('Cannot change directory to %s: %s', tmp_dir, e)
```

Comment:
Same as in backup. I  think the change to run() is enough in itself.

Reply:
Removed and tested. Seems to work just fine.